python/pfs/drp/stella/reduceArcTask.py

```diff
@@ -395,6 +409,75 @@ def updateCenters(self, fiberTraces, detectorMap):
             detectorMap.setXCenter(ft.fiberId, yy.flatten().astype(np.float32),
                                    centroids.astype(np.float32))
 
+    def tweakCentersFromArc(self, exposure, spectra, detectorMap):
+        """Update the xCenter values in the detectorMap from the arc image
+
+        We shift each xCenter spline by the average shift for that fiber.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Arc image.
+        spectra : `pfs.drp.stella.SpectrumSet`
+            Extracted spectra. We don't use the actual spectra, but this has
+            both the ``fiberId`` and reference line lists.
+        detectorMap : `pfs.drp.stella.DetectorMap`
+            Mapping of wl,fiber to detector position.
+        """
+        width, height = exposure.getDimensions()
+        badBitMask = exposure.mask.getPlaneBitMask(self.config.arcTweakMask)
+        for ss in spectra:
```

Comment:
So each fiber is shifted independently of one another? Ok, I guess a consistent update to a detectormap both spatially and spectrally is the topic of future work.